src/components/Item.js

```diff
@@ -32,14 +35,22 @@ function Item({ item, userToken }) {
   const updateLastPurchased = async (event) => {
     const docRef = doc(db, 'users', `${userToken}`, 'list', item.id);
     if (event.target.checked) {
-      await updateDoc(docRef, {
+      updateDoc(docRef, {
         lastPurchased: serverTimestamp(),
+        numberOfPurchases: item.numberOfPurchases + 1,
+        daysUntilNextPurchase: calculateEstimate(
+          item.purchaseInterval,
+          daysSincePurchased,
+          item.numberOfPurchases,
+        ),
       });
     } else {
-      await updateDoc(docRef, {
-        lastPurchased: null,
+      updateDoc(docRef, {
```

Comment:
I noticed an issue when trying this feature out on existing items in my list. This is totally non-blocking imo, I'm only mentioning in case anyone else notices this too.

For our purposes, I believe this non-blocking because we can simply reseed the database with new items. Most importantly, our users will not experience this issue because they will not be able to add items to their list without all the fields present when it is first added to the DB. 

On line 48 `updateDoc` sets the value of `daysUntilNextPurchase` and `numberOfPurchases` to NAN if the item already existed in the DB without these new fields ([screenshot](https://share.getcloudapp.com/9ZuRgyP9)). It throws this error: 

`Uncaught (in promise) FirebaseError: Function updateDoc() called with invalid data. Unsupported field value: undefined (found in field numberOfPurchases in document users/chat bonus vague/list/C80bPoqUrsknNFXmVguW)`